storage3/_async/file_api.py

```diff
@@ -39,15 +39,17 @@ async def _request(
         files: Optional[Any] = None,
         **kwargs: Any,
     ) -> Response:
-        response = await self._client.request(
-            method, url, headers=headers or {}, json=json, files=files, **kwargs
-        )
         try:
+            response = await self._client.request(
+                method, url, headers=headers or {}, json=json, files=files, **kwargs
+            )
             response.raise_for_status()
         except HTTPError:
-            raise StorageException(
-                {**response.json(), "statusCode": response.status_code}
-            )
+            try:
+                resp = response.json()
+                raise StorageException({**resp, "statusCode": response.status_code})
+            except JSONDecodeError:
+                raise StorageException({"statusCode": response.status_code})
```

Comment:
Considered adding 

`"error": "unknown error occurred"`

but didn't want to put in a field here just yet. Prefer to revisit on Mon but also open to suggestions  if anyone has ideas on how best to return this error